osu.Game.Tests/Visual/Settings/TestSceneTabletSettings.cs

```diff
@@ -17,29 +18,93 @@ namespace osu.Game.Tests.Visual.Settings
     [TestFixture]
     public class TestSceneTabletSettings : OsuTestScene
     {
-        [BackgroundDependencyLoader]
-        private void load(GameHost host)
-        {
-            var tabletHandler = new TestTabletHandler();
+        private TestTabletHandler tabletHandler;
+        private TabletSettings settings;
 
-            AddRange(new Drawable[]
+        [SetUpSteps]
+        public void SetUpSteps()
+        {
+            AddStep("create settings", () =>
             {
-                new TabletSettings(tabletHandler)
+                tabletHandler = new TestTabletHandler();
+
+                Children = new Drawable[]
                 {
-                    RelativeSizeAxes = Axes.None,
-                    Width = SettingsPanel.PANEL_WIDTH,
-                    Anchor = Anchor.TopCentre,
-                    Origin = Anchor.TopCentre,
-                }
+                    settings = new TabletSettings(tabletHandler)
+                    {
+                        RelativeSizeAxes = Axes.None,
+                        Width = SettingsPanel.PANEL_WIDTH,
+                        Anchor = Anchor.TopCentre,
+                        Origin = Anchor.TopCentre,
+                    }
+                };
             });
 
+            AddStep("set square size", () => tabletHandler.SetTabletSize(new Vector2(100, 100)));
+        }
+
+        [Test]
+        public void TestVariousTabletSizes()
+        {
             AddStep("Test with wide tablet", () => tabletHandler.SetTabletSize(new Vector2(160, 100)));
             AddStep("Test with square tablet", () => tabletHandler.SetTabletSize(new Vector2(300, 300)));
             AddStep("Test with tall tablet", () => tabletHandler.SetTabletSize(new Vector2(100, 300)));
             AddStep("Test with very tall tablet", () => tabletHandler.SetTabletSize(new Vector2(100, 700)));
             AddStep("Test no tablet present", () => tabletHandler.SetTabletSize(Vector2.Zero));
         }
 
+        [Test]
+        public void TestRotationValidity()
+        {
+            AddAssert("area valid", () => settings.AreaSelection.IsWithinBounds);
+
+            AddStep("rotate 90", () => tabletHandler.Rotation.Value = 90);
+            ensureValid();
+
+            AddStep("rotate 180", () => tabletHandler.Rotation.Value = 180);
+
+            ensureValid();
+
+            AddStep("rotate 270", () => tabletHandler.Rotation.Value = 270);
+
+            ensureValid();
+
+            AddStep("rotate 360", () => tabletHandler.Rotation.Value = 360);
+
+            ensureValid();
+
+            AddStep("rotate 0", () => tabletHandler.Rotation.Value = 0);
+            ensureValid();
+
+            AddStep("rotate 0", () => tabletHandler.Rotation.Value = 45);
```

Comment:
Is this a leftover when copy-pasting?
```suggestion
            AddStep("rotate 45", () => tabletHandler.Rotation.Value = 45);
```

Reply:
i had a change fixing this but forgot to push it. will do the next time i can, unless preempted.